platforms/hyperledger-indy/configuration/roles/create/imagepullsecret/tasks/main.yaml

```diff
@@ -15,4 +16,8 @@
 - name: Create the docker pull registry secret for {{ component_ns }}
   shell: |
     KUBECONFIG={{ kubernetes.config_file }} kubectl create secret -n {{ component_ns }} docker-registry regcred --docker-server={{ network.docker.url }} --docker-username={{ network.docker.username }} --docker-password={{ network.docker.password }} --docker-email="test@abc.mail"
-  when: secret_present.resources|length == 0
\ No newline at end of file
+  when: (secret_present.resources|length == 0) && (network.docker.username | default('', true) | trim != '') && (network.docker.password | default('', true) | trim != '')
```

Comment:
This looks okay except following:
1. `&&` to be replaced by `and` or use the conditions as an array
2. extra lines and no newline at end of file